### ISSUE TITLE
Domains for Properties with Range of Process Profile

### ISSUE THREAD

**Issue by jeannekitchens** [CredentialEngine/Schema-Development#943]
The following properties are for including Process Profiles with Resources:
[ceterms:administrationProcess](https://credreg.net/ctdl/terms/administrationProcess) [ceterms:appealProcess](https://credreg.net/ctdl/terms/appealProcess) [ceterms:complaintProcess](https://credreg.net/ctdl/terms/complaintProcess) [ceterms:developmentProcess](https://credreg.net/ctdl/terms/developmentProcess) [ceterms:maintenanceProcess](https://credreg.net/ctdl/terms/maintenanceProcess) [ceterms:reviewProcess](https://credreg.net/ctdl/terms/reviewProcess) [ceterms:revocationProcess](https://credreg.net/ctdl/terms/revocationProcess)

Currently the above properties only have Domain Includes for types of credentials.

This is not correct.  This needs to be fixed.  Many other primary classes need to  support including Process Profiles.

Assessment Profile
Learning Opportunity Profile
Learning Program
Course
Rubric
Collection
Pathway
Job
Occupation
Work Role
Task
etc...

**Comment by siuc-nate** [comment]
Currently:

Assessment Profile was created before we had established a pattern of github issues, so I can't look far enough into its history to tell why, but it (only) has:
- administrationProcess
- developmentProcess
- maintenanceProcess

Learning Opportunity Profile (likewise) was also created before we did everything through github, so I'm not sure why it doesn't have any of the process properties.  The issue that created Learning Program and updated Course ( #730 ) just specifies that both of them would [have all of the same properties](https://github.com/CredentialEngine/Schema-Development/issues/730#issuecomment-887728630) as Learning Opportunity Profile, so it looks like we didn't go into depth/detail on those at the time (or at least, we didn't document it in the issue).

Rubric only has administrationProcess, but this is [exactly what is specified](https://github.com/CredentialEngine/Schema-Development/issues/892#issuecomment-1710691870) in the issue that created it ( #892 ).

Collection doesn't have any, but this is what is specified in the issue that created it ( #793 ) and I don't see any subsequent issues that would've added those terms to Collection.

Pathway doesn't have any, but none are specified in the issue that created it ( #546 ), nor does it appear to come up in subsequent issues.

Ditto for Occupation, Job, Work Role, and Task.  #701 doesn't specify adding process to any of them, and I can't find any issues that indicate those properties should be added to those classes.

In short: unless I am missing something, the current schema with regard to process-related properties appears to be in line with what we've documented, as best I can tell.

We should probably recharacterize this issue as proposing to add those properties to those classes (is there a specific list to go with the "etc"?).

**Comment by siuc-nate** [comment]
Per today's meeting:
Update the definitions of the various process properties to refer to a resource rather than to Credential:
- appealProcess
  - Old: Formal process for objecting to decisions of the organization regarding credentials, assessments or processes.
  - New: Formal process for objecting to decisions of the organization regarding resources.
- complaintProcess
  - Old: Process for handling complaints about a credential, or aspects of it including related learning opportunities and assessments.
  - New: Process for handling complaints about a a resource, or aspects of it including related resources.
- developmentProcess
  - Old: Entity describing the process by which a credential, or aspects of it, were created.
  - New: Entity describing the process by which a resource, or aspects of it, were created.
- maintenanceProcess
  - Old: Entity describing the process by which the credential is maintained including review and updating.
  - New: Entity describing the process by which the resource is maintained, including review and updating.
- reviewProcess
  - Old: Entity that describes the process by which the credential, or aspects of it, are reviewed.
  - New: Entity that describes the process by which the resource, or aspects of it, are reviewed.

Add to Assessment Profile:
- reviewProcess
- complaintProcess

Add to Learning Opportunity Profile and its subclasses:
- developmentProcess
- maintenanceProcess
- reviewProcess
- complaintProcess

Add to Rubric:
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Collection:
- developmentProcess
- maintenanceProcess
- reviewProcess
- complaintProcess

Add to Pathway:
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Occupation:
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Job:
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Work Role:
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Task:
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to CredentialingAction and its subclasses:
- administrationProcess

Add to Condition Manifest:
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Cost Manifest:
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Pathway Set:
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Support Service:
- administrationProcess
- complaintProcess
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Transfer Intermediary:
- appealProcess
- complaintProcess
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Transfer Value Profile:
- administrationProcess
- appealProcess
- complaintProcess
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Verification Service Profile:
- administrationProcess

Add to Competency Framework:
- developmentProcess
- maintenanceProcess
- reviewProcess

Add to Competency:
- developmentProcess
- maintenanceProcess
- reviewProcess

**Comment by siuc-nate** [comment]
To be determined:
- Which Action classes should have which properties
- Should Process Profile have a CTID (e.g. for use with multiple Competencies, multiple Transfer Value Profiles, etc.)

**Comment by philbarker** [comment]
Overall looks good to me. I would suggest some tweaks to the definitions provided [above](https://github.com/CredentialEngine/Schema-Development/issues/943#issuecomment-2245741688). 

I don't think saying "about a resource" or the like adds much when "a resource" can be anything. 

The definitions also alternate between referring to subject as "**the** organization" "**a** resource" etc. I don't know if this is for a reason. I prefer definitions that don't assume anything when writing term definitions/data dictionaries. They should work in the context of interpreting a graph database (e.g. how to find all the things that use this process), rather than assuming you're dealing with just one thing: so have defaulted to "a" (I think Nate may differ, I don't mind much which is used so long as it is consistent). 

Some definitions refer to a "Process" others to an "Entity that describes a process". They all point to ProcessProfiles, which are descriptions, not the process itself, but I don't think repeating "Entity" here makes anything easy to understand.

Q: is it significant that only the appeal process need be Formal? 

So, for:

> appealProcess
> 
>  * Old: Formal process for objecting to decisions of the organization regarding credentials, assessments or processes.
>  * New: Formal process for objecting to decisions of the organization regarding resources.
> 

I suggest
 * New: Description of a formal process for objecting to decisions of an organization.

for 

> complaintProcess
> 
> * Old: Process for handling complaints about a credential, or aspects of it including related learning opportunities and assessments.
> * New: Process for handling complaints about a a resource, or aspects of it including related resources.
> 

I suggest 
* New: Description of a  process for handling complaints about a resource or related resources.

(note "a a resource" in original; I think "aspects of a resource" are inherent in "about a resource" and muddies the reference to related resource.)

> developmentProcess
> 
>  * Old: Entity describing the process by which a credential, or aspects of it, were created.
>  * New: Entity describing the process by which a resource, or aspects of it, were created.
> 

I suggest:
* New: Description of a  process by which a resource was created.

(I could be persuaded to keep the "or aspects of it" in this case if the next definition also referenced maintaining "aspects"... but I don't think it's needed)

> maintenanceProcess
> 
> * Old: Entity describing the process by which the credential is maintained including review and updating.
> * New: Process by which the resource is maintained, including review and updating.
> 

I suggest
*   New: Description of a process by which a resource is maintained, including review and updating.

> reviewProcess
> 
>  * Old: Entity that describes the process by which the credential, or aspects of it, are reviewed.
>  * New: Entity that describes the process by which the resource, or aspects of it, are reviewed.

I suggest:
*  New: Description of a process by which a resource is reviewed.


Did you consider including the new EEOM classes in the Domains?

**Comment by siuc-nate** [comment]
I'm fine with those definitions. @jeannekitchens what do you think? 

>Did you consider including the new EEOM classes in the Domains?

Currently the only item there is to add administrationProcess to Data Set Profile. Are there others we need?

**Comment by siuc-nate** [comment]
Here's the current proposal (I will update this if something changes):

**Update Existing Properties**
> **URI:** ceterms:administrationProcess
> **Add Domain:** ceterms:CredentialingAction (and subclasses), ceterms:SupportService, ceterms:TransferValueProfile, ceterms:VerificationServiceProfile

> **URI:** ceterms:appealProcess
> **Remove Definition:** Formal process for objecting to decisions of the organization regarding credentials, assessments or processes.
> **Add Definition:** Description of a formal process for objecting to decisions of an organization.
> **Add Domain:** ceterms:TransferIntermediary, ceterms:TransferValueProfile

> **URI:** ceterms:complaintProcess
> **Remove Definition:** Process for handling complaints about a credential, or aspects of it including related learning opportunities and assessments.
> **Add Definition:** Description of a process for handling complaints about a resource or related resources.
> **Add Domain:** ceterms:AssessmentProfile, ceterms:LearningOpportunityProfile (and subclasses), ceterms:Collection, ceterms:SupportService, ceterms:TransferIntermediary, ceterms:TransferValueProfile

> **URI:** ceterms:developmentProcess
> **Remove Definition:** Entity describing the process by which a credential, or aspects of it, were created.
> **Add Definition:** Description of a process by which a resource was created.
> **Add Domain:** ceterms:LearningOpportunityProfile (and subclasses), ceterms:Rubric, ceterms:Collection, ceterms:Pathway, ceterms:Occupation, ceterms:Job, ceterms:WorkRole, ceterms:Task, ceterms:ConditionManifest, ceterms:CostManifest, ceterms:PathwaySet, ceterms:SupportService, ceterms:TransferIntermediary, ceterms:TransferValueProfile, ceasn:CompetencyFramework, ceasn:Competency

> **URI:** ceterms:maintenanceProcess
> **Remove Definition:** Entity describing the process by which the credential is maintained including review and updating.
> **Add Definition:** Description of a process by which a resource is maintained, including review and updating.
> **Add Domain:** ceterms:LearningOpportunityProfile (and subclasses), ceterms:Rubric, ceterms:Collection, ceterms:Pathway, ceterms:Occupation, ceterms:Job, ceterms:WorkRole, ceterms:Task, ceterms:ConditionManifest, ceterms:CostManifest, ceterms:PathwaySet, ceterms:SupportService, ceterms:TransferIntermediary, ceterms:TransferValueProfile, ceasn:CompetencyFramework, ceasn:Competency

> **URI:** ceterms:reviewProcess
> **Remove Definition:** Entity that describes the process by which the credential, or aspects of it, are reviewed.
> **Add Definition:** Description of a process by which a resource is reviewed.
> **Add Domain:** ceterms:AssessmentProfile, ceterms:LearningOpportunityProfile (and subclasses), ceterms:Rubric, ceterms:Collection, ceterms:Pathway, ceterms:Occupation, ceterms:Job, ceterms:WorkRole, ceterms:Task, ceterms:ConditionManifest, ceterms:CostManifest, ceterms:PathwaySet, ceterms:SupportService, ceterms:TransferIntermediary, ceterms:TransferValueProfile, ceasn:CompetencyFramework, ceasn:Competency

**Comment by philbarker** [comment]
> >    Did you consider including the new EEOM classes in the Domains?
>
>Currently the only item there is to add administrationProcess to Data Set Profile. Are there others we need?
>

If we a looking for completeness and consistency and want to avoid having to go over this again later, I would add developmentProcess, reviewProcess and maintenanceProcess to DataSetProfile. 

>Any updates needed to the definition for administrationProcess?

Good point! Currently we have

> **Definition:** Entity describing the process by which a resource, or aspects of it, are administered. 

An edit along the lines I suggested above would give

> **Definition:** Description of a process by which a resource is administered. 

(feel free to add back in the "or aspects of it" if you think there are cases where that matters)

**Comment by siuc-nate** [comment]
Thanks, Phil. I added your suggestions to my previous post. @jeannekitchens can you or Phil think of anything else, or is this issue ready to go?

**Comment by philbarker** [comment]
>  can you or Phil think of anything else, or is this issue ready to go?

I've given it all I've got, Looks good enough to me.

**Comment by siuc-nate** [comment]
These changes have been made in pending CTDL and noted in the history tracking.
